src/tests/ftest/control/config_generate_output.py

```diff
@@ -0,0 +1,447 @@
+                shortest = len(self.nvme_socket_to_addrs[socket_id])
+                shortest_id = socket_id
+
+        min_ssd = len(self.nvme_socket_to_addrs[shortest_id])
+        self.log.info("Maximum --min-ssds threshold = %d", min_ssd)
+
+        dmg = self.get_dmg_command()
+        errors = []
+
+        # Call dmg config generate --min-ssds=<1 to min_ssd>. Should pass.
+        for num_ssd in range(1, min_ssd + 1):
+            try:
+                dmg.config_generate(access_points="wolf-a", min_ssds=num_ssd)
+            except CommandFailure:
+                errors.append(
+                    "config generate failed with min_ssd = {}!".format(num_ssd))
+
+        # Call dmg config generate --min_ssds=<min_ssd + 1>. Should fail.
+        try:
+            dmg.config_generate(access_points="wolf-a", min_ssds=min_ssd + 1)
+            errors.append(
+                "config generate succeeded with min_ssd + 1 = {}!".format(
+                    min_ssd + 1))
+        except CommandFailure:
+            self.log.info(
+                "config generated failed with min_ssd + 1 as expected")
+
+        # Call dmg config generate --min-ssds=0
+        generated_yaml = dmg.config_generate(access_points="wolf-a", min_ssds=0)
+        # Iterate the engines and verify that there's no bdev_list field.
+        engines = generated_yaml["engines"]
+        for engine in engines:
+            if "bdev_list" in engine:
+                errors.append("bdev_list field exists with --min-ssds=0!")
+
+        self.check_errors(errors)
+
+    def test_net_class(self):
+        """Test --net-class.
+
+        1. Iterate the interface set and count the number of elements that
+        starts with "ib". This would be our ib_count threshold that we can set
+        --num-engines with --net-class=infiniband.
+        2. Call dmg config generate --net-class=infiniband
+        --num-engines=<1 to ib_count> and verify that it works.
+        3. In addition, verify provider using the dictionary. i.e., iterate
+        "engines" fields and verify "provider" is in the list where key is 
+        "fabric_iface".
+        4. Similarly find eth_count and call dmg config generate
+        --net-class=ethernet --num-engines=<1 to eth_count> and verify that it
+        works.
+        5. As in ib, also verify provider using the dictionary. i.e., iterate
+        "engines" fields and verify "provider" is in the list where key is
+        "fabric_iface".
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,small
+        :avocado: tags=control,config_generate_entries,net_class
+        """
+        # Get necessary storage and network info.
+        self.prepare_expected_data()
+
+        # Get ib_count threshold.
+        ib_count = 0
+        for interface in self.interface_set:
+            if interface[:2] == "ib":
+                ib_count += 1
+        self.log.info("ib_count = {}".format(ib_count))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions